base/sparse/sparsematrix.jl

```diff
@@ -2896,7 +2896,20 @@ function hcat(X::SparseMatrixCSC...)
     SparseMatrixCSC(m, n, colptr, rowval, nzval)
 end
 
-function hvcat(rows::Tuple{Vararg{Int}}, X::SparseMatrixCSC...)
+
+# Sparse/dense concatenation
+
+function hcat(Xin::Union{Matrix, SparseMatrixCSC}...)
+    X = SparseMatrixCSC[issparse(x) ? x : sparse(x) for x in Xin]
```

Comment:
oh yeah I guess we could technically use a generator here now, though I'm not sure if it would make a noticeable difference